js/scripts.js

```diff
@@ -0,0 +1,118 @@
+// Variables
+var newGameBtn = document.getElementById('js-newGameButton');
+var pickRock = document.getElementById('js-playerPick_rock');
+var pickPaper = document.getElementById('js-playerPick_paper');
+var pickScissors = document.getElementById('js-playerPick_scissors');
+var newGameElem = document.getElementById('js-newGameElement');
+var pickElem = document.getElementById('js-playerPickElement');
+var resultsElem = document.getElementById('js-resultsTableElement');
+var playerPointsElem = document.getElementById('js-playerPoints');
+var playerNameElem = document.getElementById('js-playerName');
+var computerPointsElem = document.getElementById('js-computerPoints');
+var playerPickElem = document.getElementById('js-playerPick');
+var computerPickElem = document.getElementById('js-computerPick');
+var playerResultElem = document.getElementById('js-playerResult');
+var computerResultElem = document.getElementById('js-computerResult');
+
+var gameState = 'notStarted';	//started, ended
+var player = {
+	name: '',
+	score: 0
+};
+var computer = {
+	score: 0
+};
+
+// Events
+newGameBtn.addEventListener('click', newGame);
+pickRock.addEventListener('click', function() { playerPick('kamień') });
+pickPaper.addEventListener('click', function() { playerPick('papier') });
+pickScissors.addEventListener('click', function() { playerPick('nożyce') });
+
+// Functions
+function setGameElements() {
+	switch(gameState) {
+		case 'started':
+			newGameElem.style.display = 'none';
+			pickElem.style.display = 'block';
+			resultsElem.style.display = 'block';
+		break;
+		case 'ended':
+			newGameBtn.innerText = 'Jeszcze raz';
+		case 'notStarted':
+		default:
+			newGameElem.style.display = 'block';
+			pickElem.style.display = 'none';
+			resultsElem.style.display = 'none';
+	}
+}
+
+function newGame() {
+	player.name = prompt('Graczu, podaj swoje imię', 'Twoje imię');
+	if (player.name) {
+		player.score = computer.score = 0;
+		gameState = 'started';
+		setGameElements();
+
+		playerNameElem.innerHTML = player.name;
+		setGamePoints();
+	}
+}
+
+function getComputerPick() {
+	var possiblePicks = ['kamień', 'papier', 'nożyce'];
+	return possiblePicks[Math.floor(Math.random()*3)];
+}
+
+function checkRoundWinner(playerPick, computerPick) {
```

Comment:
Zauważ, że w tej funkcji robimy trochę masło maślane. Na podstawie picku gracza i komputera uzupełniamy zmienną `winnerIs`, a potem… sprawdzamy tą zmienną i zależnie od jej wartości podejmujemy akcje.

Spróbuj przerobić to tak żebyśmy zamiast wypełniać zmienną `winnerIs` - od razu podejmowali odpowiednie akcje. Te akcje najlepiej ubrać w nową funkcję bo w obecym stanie rzeczy ona robi zbyt wiele rzeczy na raz. Teoretycznie funkcja powinna robić maksymalnie jedną rzecz (zasada single responsibility)